### DISCUSSION TITLE
How do i catch errors throw by the modal?

### DISCUSSION THREAD

**Discussion by lunarinlunarinlunarin** [rainbow-me/rainbowkit#1456] [Q&A]
As stated in the title. I cannot find a way to catch the errors being thrown by the modal. I would like to at least be able to handle these exceptions and not crash my dapp

**Comment by DanielSinclair** [comment]
@lunarinlunarinlunarin What kind of exceptions are you seeing?

**Comment by einaralex** [comment]
I'm having the same issue, when I click WalletConnect, I get the following error: `'No matching key. keychain: 1b3c116616f65fd2ca8691447b3282af7bbff985cd29879484be367edfdeec26'` 
I know the solution is to clear the localStorage, but I would like to handle it for the customer if they run into this error.

**Reply by 0xmkh** [reply]
@einaralex We can probably expose a hook where you can get `onWalletConnectError`, `onSwitchChainError` and more, but would have to look at this internally before doing the decision.

Could you tell me how you've got that error from WalletConnect ? Best if you can link us a small reproducible example.

**Reply by einaralex** [reply]
You can manually enforce this state by connecting a WalletConnect wallet, go into DevTools>Application>IndexedDB>WALLET_CONNECT_V2_INDEXED_DB>keyvaluestorage 
Delete `wc@2:core:0.3:keychain`, then trigger a signature request and the error should appear.

**Reply by 0xmkh** [reply]
Alright that sounds good. Will keep this discussion open and let you know if there will be any updates to this 👍
